client/src/components/Seat/index.js

```diff
@@ -1,23 +1,36 @@
-import React, { useState } from 'react';
+import React, { useState, useContext } from 'react';
 import './style.scss';
+import { ReservationContext } from '../../context/Reservation';
 
-const Seat = () => {
-  const [selected, setSelected] = useState(false);
+const Seat = ({ columnNr, seatNr }) => {
+  const [selectedSeats, setSelectedSeats, tickets, setTickets] = useContext(
+    ReservationContext,
+  );
   // TO DO gest variables from props
   const occupied = false;
-  const seatNr = 1;
   // TO DO move the handle function to the parent component
   const handleSeatSelected = (e) => {
     e.target.classList.toggle('selected');
-    setSelected((prevSelected) => !prevSelected);
+    if (selectedSeats.includes(seatNr)) {
+      const newSelectedSeats = selectedSeats.filter((seat) => {
+        return seat !== seatNr;
+      });
+      setSelectedSeats((prevselectedSeats) => [newSelectedSeats]);
+      setTickets((prevTickets) => prevTickets - 1);
+    } else {
+      setSelectedSeats((prevSelectedSeats) => [...prevSelectedSeats, seatNr]);
+      setTickets((prevTickets) => prevTickets + 1);
```

Comment:
There is some bug causing the first element in the array become an array itself. Check and see if you can fix it.
![image](https://user-images.githubusercontent.com/58532638/113433316-42a48380-93df-11eb-910f-a108defaa0fd.png)